test/common/formatter/substitution_formatter_test.cc

```diff
@@ -2092,6 +2095,113 @@ TEST(SubstitutionFormatterTest, FilterStateFormatter) {
   }
 }
 
+TEST(SubstitutionFormatterTest, ProxyProtocolTlvsFormatter) {
+  Http::TestRequestHeaderMapImpl request_headers{{":method", "GET"}, {":path", "/"}};
+  Http::TestResponseHeaderMapImpl response_headers;
+  Http::TestResponseTrailerMapImpl response_trailers;
+  std::string body;
+  auto src_addr =
+      Network::Address::InstanceConstSharedPtr(new Network::Address::Ipv4Instance("1.2.3.4", 773));
+  auto dst_addr =
+      Network::Address::InstanceConstSharedPtr(new Network::Address::Ipv4Instance("0.1.1.2", 513));
+
+  // No parameter value passed in for tlv_type. 
+  {
+    EXPECT_THROW_WITH_MESSAGE(SubstitutionFormatParser::parse("%PROXY_PROTOCOL_TLVS()%"), EnvoyException,
+                              "PROXY_PROTOCOL_TLVS requires parameters");
+  }
+
+  // Invalid tlv_type passed as parameter -- non-integer
+  {
+    EXPECT_THROW_WITH_MESSAGE(SubstitutionFormatParser::parse("%PROXY_PROTOCOL_TLVS(INVALID_TLV_TYPE)%"), EnvoyException,
+                              "Invalid parameter provided for PROXY_PROTOCOL_TLVS header: INVALID_TLV_TYPE. Not parsable as int.");
+  }
+
+  // Invalid tlv_type passed as parameter -- too large of a number for stoi to parse as int
+  {
+    EXPECT_THROW_WITH_MESSAGE(SubstitutionFormatParser::parse("%PROXY_PROTOCOL_TLVS(3000000000)%"), EnvoyException,
+                              "Invalid parameter provided for PROXY_PROTOCOL_TLVS header: 3000000000. Not parsable as int.");
+  }
+
+  // Invalid tlv_type passed as parameter -- too large of a number to be a valid TLV type
+  {
+    EXPECT_THROW_WITH_MESSAGE(SubstitutionFormatParser::parse("%PROXY_PROTOCOL_TLVS(300)%"), EnvoyException,
+                              "Invalid parameter provided for PROXY_PROTOCOL_TLVS header: 300. Not a valid TLV type.");
+  }
+
+  // ProxyProtocolFilterState is not stored in FilterState
+  {
+    NiceMock<StreamInfo::MockStreamInfo> stream_info;
+
+    EXPECT_CALL(Const(stream_info), filterState()).Times(testing::AtLeast(1));
+
+    auto providers = SubstitutionFormatParser::parse("%PROXY_PROTOCOL_TLVS(2)%");
+
```

Comment:
super nit: I feel removing these blank lines in the same test case makes it easier to read. I know this file isn't consistent on that. Here's the [Google style guide](https://google.github.io/styleguide/cppguide.html#Vertical_Whitespace).